frontend/src/providers/QueryBuilder.tsx

```diff
@@ -867,6 +880,12 @@ export function QueryBuilderProvider({
 
 	const handleRunQuery = useCallback(
 		(shallUpdateStepInterval?: boolean, newQBQuery?: boolean) => {
+			const isExplorer =
+				location.pathname === ROUTES.LOGS_EXPLORER ||
+				location.pathname === ROUTES.TRACES_EXPLORER;
+			if (isExplorer) {
+				setHasRunOnExplorer(true);
```

Comment:
Please help me understand what's going on here. It says handle run query and set the `HasRunOnExplorer` to true.